### PR TITLE
Feature/swift5 migration

### PR DESCRIPTION
**Pull request by earlgaspard**

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 line_length: 245
 
 disabled_rules:
+  - identifier_name # Disables SwiftLint complaining about enums with associated values
```
Comment (wmcginty) [inline .swiftlint.yml]: This got fixed in a new version of SwiftLint. If it's a local version that's running, updating should fix.

Comment (earlgaspard) [reply .swiftlint.yml]: I did update my local version of SwiftLint to 0.25.1 and was still seeing this. I'll check again

File: Sources/SessionTools/Base/Session.swift [hunk 2]
```diff
@@ -31,7 +31,7 @@ open class Session<T: Codable> {
     }
     
     public var currentItem: T? {
-        guard let user = try? container.item(forIdentifier: storageIdentifier, jsonDecoder: decoder) else { return nil }
+        guard let user = ((try? container.item(forIdentifier: storageIdentifier, jsonDecoder: decoder)) as T??) else { return nil }
```
Comment (tylermilner) [inline Sources/SessionTools/Base/Session.swift]: The double optional feels weird here. Is there anything we can do to try and make this simpler?

Comment (earlgaspard) [reply Sources/SessionTools/Base/Session.swift]: I am not a fan of it either but that's what the migrator recommended. I prefer the old way but I'll see if there is a simpler way.

Comment (earlgaspard) [reply Sources/SessionTools/Base/Session.swift]: I changed it back and it seems to work.

Comment (tylermilner) [reply Sources/SessionTools/Base/Session.swift]: Actually, we may need to use the `swift_version` attribute inside of the Podspec. Looks like `.swift-version` file support is [going away](https://stackoverflow.com/a/54060826/4343618) in Cocoapods 1.7.0.

Comment (earlgaspard) [reply Sources/SessionTools/Base/Session.swift]: Going to update the pod spec

**Comment by earlgaspard** [comment]
@tylermilner It worked.

**Pull request merged**